### PR TITLE
fix: multiple locales merge in nested case.

### LINKED ISSUE #10: remove "merge-descriptors" dependency at all?
Hi -

this time no bug, only an idea:
Since there is Node v4.0 out, would it make sense to replace the line
```javascript
merge(resources[locale], resource);
```
in index.js, line 62 with 
```javascript
Object.assign(resources[locale], resource);
```
?

### PR DESCRIPTION
**Pull request by huacnlee**
- Removed merge-merge-descriptors npm;
- Add test case with nested merge.

And fixed #10

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -219,4 +218,18 @@ module.exports = function (app, options) {
...
     return data;
   }
+
+  // Deep merge nested keys, source key will override target.
+  function deepMerge(target, source) {
+    target = target || {};
+    source = source || {};
+
+    for(var key in source) {
+      if (typeof target[key] === 'object') {
```
Comment (fengmk2) [inline index.js]: `typeof null` will be `object` too.

Comment (huacnlee) [inline index.js]: Is this right `typeof target[key] === 'object' && target[key] !== null`?

**Comment by hotoo** [comment]
SEE #12 FIRST PLEASE, THIS PR MAYBE CAN CLASED.

**Comment by fengmk2** [comment]
see #12

**Pull request closed without merging**